### PR TITLE
Add fast route for retirement

### PR DESCRIPTION
**Pull request by marten**

### CODE REVIEW

File: spec/controllers/api/v1/workflows_controller_spec.rb [hunk 1]
```diff
@@ -508,6 +508,21 @@
...
+      default_request scopes: scopes, user_id: authorized_user.id
+    end
+
+    let(:subject_set_project) { project }
+    let(:subject_set) { create(:subject_set, project: project, workflows: [project.workflows.first]) }
+    let(:subject_set_id) { subject_set.id }
+    let(:subject) { create(:subject, subject_sets: [subject_set]) }
+
+    it 'retires the subject' do
+      post :retired_subjects, workflow_id: workflow.id, subject_id: subject.id
```
Comment (marten) [inline spec/controllers/api/v1/workflows_controller_spec.rb]: The route works fine in dev server, but tests cant find it. @camallen @parrish any ideas?

File: app/controllers/api/v1/workflows_controller.rb [hunk 2]
```diff
@@ -24,6 +24,11 @@ def destroy_links
     super { |workflow| reload_queue(workflow) }
   end
 
+  def retire_subject
+    controlled_resource.retire_subject(params[:subject_id])
+    render nothing: true
```
Comment (camallen) [inline app/controllers/api/v1/workflows_controller.rb]: what is the response status from this?

Comment (marten) [inline app/controllers/api/v1/workflows_controller.rb]:
```
HTTP/1.1 200 OK
X-Frame-Options: SAMEORIGIN
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
Content-Type: text/plain; charset=utf-8
P3P: CP="NOI ADM DEV PSAi COM NAV OUR OTRo STP IND DEM"
Cache-Control: no-cache
X-Request-Id: b95a2bda-732c-4a24-95dc-7e114d5d5290
X-Runtime: 0.036305
Vary: Origin
Connection: close
Transfer-Encoding: chunked

```

File: app/controllers/api/v1/workflows_controller.rb [hunk 3]
```diff
@@ -24,6 +24,11 @@ def destroy_links
     super { |workflow| reload_queue(workflow) }
   end
 
+  def retire_subject
```
Comment (camallen) [inline app/controllers/api/v1/workflows_controller.rb]: will this accept multiple subject ids? if so maybe the name should be `retire_subjects`?

File: app/controllers/api/v1/workflows_controller.rb (hunk 2 above)
Comment (marten) [inline app/controllers/api/v1/workflows_controller.rb]: Nero doesn't need more than the 200, and rendering the workflow added 150ms of overhead and a bunch of queries, that's why I added this line in.

Comment (camallen) [inline app/controllers/api/v1/workflows_controller.rb]: sounds good.

File: spec/controllers/api/v1/workflows_controller_spec.rb [hunk 4]
```diff
@@ -508,6 +508,21 @@
...
+      default_request scopes: scopes, user_id: authorized_user.id
+    end
+
+    let(:subject_set_project) { project }
+    let(:subject_set) { create(:subject_set, project: project, workflows: [project.workflows.first]) }
+    let(:subject_set_id) { subject_set.id }
+    let(:subject) { create(:subject, subject_sets: [subject_set]) }
+
+    it 'retires the subject' do
+      post :retire_subject, workflow_id: workflow.id, subject_id: subject.id
```
Comment (camallen) [inline spec/controllers/api/v1/workflows_controller_spec.rb]: what is the expectation here?

Comment (marten) [inline spec/controllers/api/v1/workflows_controller_spec.rb]: i was so happy the route finally worked i totally forgot about that one...

File: app/controllers/api/v1/workflows_controller.rb (hunk 3 above)
Comment (marten) [inline app/controllers/api/v1/workflows_controller.rb]: No, just one right now.

**Comment by camallen** [comment]
LGTM - nice one.

**Pull request merged**